File: Java-SpringBoot/BE-01-InMemory/java-21-springboot-4.0/src/main/java/com/flyrank/userservice/repository/UserRepository.java
package com.flyrank.userservice.repository;

import com.flyrank.userservice.dto.UserResponse;

import java.util.Optional;
import java.util.SequencedCollection;
import java.util.UUID;

/**
 * Java 21 Repository interface showcasing SequencedCollection for ordered retrieval.
 */
public interface UserRepository {
    UserResponse save(UserResponse user);

    Optional<UserResponse> findById(UUID id);

    /**
     * Java 21 SequencedCollection interface returning stored users in insertion order.
     */
    SequencedCollection<UserResponse> findAllSequenced();
}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/repository/UserRepository.kt
package com.flyrank.userservice.repository

import com.flyrank.userservice.dto.UserResponse
import java.util.UUID

/**
 * Repository interface using idiomatic Kotlin functions.
 */
interface UserRepository {
    fun save(user: UserResponse): UserResponse
    fun findById(id: UUID): UserResponse?
    fun findAll(): List<UserResponse>
}
